components/survey_monkey/sources/custom-webhook-events/custom-webhook-events.mjs

```diff
@@ -0,0 +1,107 @@
+import surveyMonkey from "../../survey_monkey.app.mjs";
+
+export default {
+  name: "Custom webhook events",
```

Comment:
I see some code being reused in these sources. What do you think about making a `common/base.mjs` file containing code that's common to all the sources? Here's what I would do:

```
// survey_monkey/sources/common/base.mjs

import surveyMonkey from "../../survey_monkey.app.mjs";

export default {
  props: {
    surveyMonkey,
    http: "$.interface.http",
    db: "$.service.db",
  },
  hooks: {
    async deactivate() {
      await this.surveyMonkey.deleteHook(this.getHookId());
    },
  },
  methods: {
    getHookId() {
      return this.db.get('hookId');
    },
    setHookId(hookId) {
      this.db.set('hookId');
    },
  },
  async run(event) {
    this.http.respond({
      status: 200,
    });
    this.$emit(event, {
      summary: this.getSummary(event),
      ts: Date.now(),
    });
  }
}
```

It's recommended to have "getter" and "setter" methods for writing to `db`, which is what `getHookId` and `setHookId` would be for. 
The `deactivate` hook is the same for all sources and they share some common props, so we can put those here. The only difference in the `run` method is the summary, so maybe we could have a common `run` method, and each component can have their own `getSummary` method.
```suggestion
import surveyMonkey from "../../survey_monkey.app.mjs";
import common from "../common/base.mjs";

export default {
  ...common,
  name: "Custom webhook events",
```